### ISSUE TITLE
Circular referenced types will take forever to populate

### ISSUE THREAD

**Issue by engtecmat** [zhezhangz/easy-modeling#1]
Hi there, I really like these features of the library.

**I hope it can support that two objects depend on each other.**
When I have the following code, my program always creates objects till it eats all memory from my computer.

```
public class Course {
  
   Set<Student> students;
   
}
```

```
public class Student {

  Set<Course> courses;

}

```
**Is the `allowEmpty` of the annotation `Field` a solution. I’m sure I use it correctly, but it doesn't work.**
```
@Model(type = Student.class, fields = {@Field(name = "courses", allowEmpty = true)})
public class StudentConfig {
}
```

**Comment by zhezhangz** [comment]
Hello @engtecmat,
Thanks for reporting! I have reproduced this problem and confirmed this is a bug. I'm going to fix it asap.

**Comment by zhezhangz** [comment]
@engtecmat Thanks for reporting! I have reproduced this problem and confirmed it's a bug. 
I'm going to fix it asap.

**Comment by zhezhangz** [comment]
@engtecmat Thanks again for bringing up this issue! I have fixed that and released a new version of `0.1.0-Beta`. The release notes can be found [here](https://github.com/easymodeling/easy-modeling/releases/tag/0.1.0-Beta). 
Though it's still a pre-release, I believe it also could be worth a try. Please upgrade to this version if it's possible on your side.

### About `allowEmpty`

Regarding the second part of your question, the attritube `allowEmpty` in annotation `@Model` will only work for `Optional` fields. Related information will be documented later.

**Comment by engtecmat** [comment]
> @engtecmat Thanks again for bringing up this issue! I have fixed that and released a new version of `0.1.0-Beta`. The release notes can be found [here](https://github.com/easymodeling/easy-modeling/releases/tag/0.1.0-Beta). Though it's still a pre-release, I believe it also could be worth a try. Please upgrade to this version if it's possible on your side.
> 
> ### About `allowEmpty`
> Regarding the second part of your question, the attritube `allowEmpty` in annotation `@Model` will only work for `Optional` fields. Related information will be documented later.

Thank you very much for fixing briefly. And thanks for the explanation.
